### PR TITLE
Address issue #1202.

### PR DESCRIPTION
**Pull request by NiklasGustafsson**
Also: add capability to load a script module from a byte array.

### CODE REVIEW

**Comment by NiklasGustafsson** [comment]
@shaltielshmid -- could you take a quick look at this PR?

File: src/TorchSharp/NN/Upsample.cs [hunk 1]
```diff
@@ -220,6 +226,21 @@ public override Tensor forward(Tensor tensor)
...
+            public ReadOnlySpan<long> size {
+                get { return _size is null ? null : new ReadOnlySpan<long>(_size!); }
+            }
+
+            public ReadOnlySpan<double> scale_factor {
+                get { return _scale_factor is null ? null : new ReadOnlySpan<double>(_scale_factor!); }
+            }
+
+            private long[]? _size;
+            private double[]? _scale_factor;
```
Comment (shaltielshmid) [inline src/TorchSharp/NN/Upsample.cs:242]: Is there a need for a separate field for the size and scale factor? Why not just store them as regular properties like the other fields?

**Review by shaltielshmid — APPROVED** [review]

**Comment by shaltielshmid** [comment]
A general question: Do we want to expose these properties on all the classes?

For example: 
All the dropouts (e.g., `AlphaDropout`, `Dropout1d`) should expose `p` and `inplace` like in PyTorch.
`Flatten` with `start_dim` and `end_dim`
etc.

File: src/TorchSharp/NN/Upsample.cs (hunk 1 above)
Comment (NiklasGustafsson) [reply src/TorchSharp/NN/Upsample.cs:242]: The C# compiler didn't like the ReadOnlySpan<T> as the property type when the getter/setters were automatically generated.

**Comment by NiklasGustafsson** [comment]
If you put the list together, we can split the job of implementing them...

**Comment by NiklasGustafsson** [comment]
Note that there are some properties that are actually modified in native code, and for those, we have to get them with a native call. The properties on BatchNorm, for example.

**Comment by shaltielshmid** [comment]
Ah, good point! Will definitely keep that in mind.
